xrootd/protocol/signRequirements.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package protocol contains the XRootD protocol specific types.
+package protocol // import "go-hep.org/x/hep/xrootd/protocol"
+
+import (
+	"go-hep.org/x/hep/xrootd/protocol/protocol"
+)
+
+// SignRequirements implements a way to check if request should be signed
+// according to XRootD protocol specification v. 3.1.0, p.75-76.
+type SignRequirements struct {
+	requirements map[uint16]protocol.RequestLevel
+}
+
+// Needed checks if the request should be signed.
+func (sr *SignRequirements) Needed(requestID uint16, modifies bool) bool {
+	v, exist := sr.requirements[requestID]
+	if !exist || v == protocol.SignNone {
+		return false
+	}
+	if v == protocol.SignLikely && !modifies {
+		return false
+	}
+	return true
+}
+
+// NewSignRequirementsDefault creates a default SignRequirements with "None" security level.
+func NewSignRequirementsDefault() SignRequirements {
```

Comment:
to follow suit of `http.DefaultClient` or `gonum.org/v1/gonum/optimize.DefaultSettings`, I would:
`s/NewSignRequirementsDefault/DefaultSignRequirements/`